app/services/course/assessment/question/programming/java/java_package_service.rb

```diff
@@ -0,0 +1,350 @@
+# frozen_string_literal: true
+class Course::Assessment::Question::Programming::Java::JavaPackageService < \
+  Course::Assessment::Question::Programming::LanguagePackageService
+
+  def initialize(params)
+    @test_params = test_params params if params.present?
+  end
+
+  def submission_templates
+    if submit_as_file?
+      templates = []
+      @test_params[:submission_files].map do |file|
+        template_file = { filename: file.original_filename, content: File.read(file.tempfile.path) }
+        templates.push(template_file)
+      end
+
+      templates
+    else
+      [
+        filename: 'template',
+        content: @test_params[:submission] || ''
+      ]
+    end
+  end
+
+  def generate_package(old_attachment)
```

Comment:
I think only `get_files_meta` has similar functionality as the other two package services, just that it was renamed from `get_data_files_meta` so that it would make more sense, since it is supposed to handle any file type (submission/solution/data).

If were to refactor, then would pull this function into `language_package_service` and rename the functions used in the other two package services